fastmri/data/transforms.py

```diff
@@ -509,3 +509,147 @@ def __call__(
+                    from the RSS opp of all coils pre-compression.
+                fname: File name.
+                slice_num: The slice index.
+                max_value: Maximum image value.
+                crop_size: The size to crop the final image.
+        """
+        if target is not None:
+            target = to_tensor(target)
+            max_value = attrs["max"]
+        else:
+            target = torch.tensor(0)
+            max_value = 0.0
+
+        if self.crop_size is None:
+            crop_size = torch.tensor([attrs["recon_size"][0], attrs["recon_size"][1]])
+        else:
+            if isinstance(self.crop_size, tuple) or isinstance(self.crop_size, list):
+                assert len(self.crop_size) == 2
+                if self.crop_size[0] is None or self.crop_size[1] is None:
+                    crop_size = torch.tensor(
+                        [attrs["recon_size"][0], attrs["recon_size"][1]]
+                    )
+                else:
+                    crop_size = self.crop_size
+            elif isinstance(self.crop_size, int):
+                crop_size = (self.crop_size, self.crop_size)
+            else:
+                raise ValueError(
+                    f"`crop_size` should be None, tuple, list, or int, not: {type(self.crop_size)}"
+                )
+
+        if self.num_compressed_coils is None:
+            num_compressed_coils = kspace.shape[0]
+        else:
+            num_compressed_coils = self.num_compressed_coils
+
+        seed = None if not self.use_seed else tuple(map(ord, fname))
+        acq_start = 0
+        acq_end = crop_size[1]
+
+        # new cropping section
+        # why sample every other point you may ask?
+        # the answer is that the readout axis is 2-factor oversampled
+        # we do this in MRI to prevent wraparound by the anatomy if the volume
+        # selection did not perform well enough
+        kspace = kspace[:, ::2]
+        kspace = center_crop(kspace, crop_size)
+
+        # we calculate the target before coil compression. This causes the mini
+        # simulation to be one where we have a 15-coil, low-resolution image
+        # and our reconstructor has an SVD coil approximation. This is a little
+        # bit more realistic than doing the target after SVD compression
+        target = fastmri.rss_complex(fastmri.ifft2c(to_tensor(kspace)))
+        max_value = target.max()
+
+        # apply coil compression
+        new_shape = (num_compressed_coils,) + kspace.shape[1:]
+        kspace = np.reshape(kspace, (kspace.shape[0], -1))
+        left_vec, _, _ = np.linalg.svd(kspace, compute_uv=True, full_matrices=False)
+        kspace = np.reshape(
+            np.array(np.matrix(left_vec[:, :num_compressed_coils]).H @ kspace),
+            new_shape,
+        )
+        kspace = to_tensor(kspace)
+
+        # Mask kspace
+        if self.mask_func:
+            masked_kspace, mask = apply_mask(
```

Comment:
I get an error here when trying to run the first command in the README, since `apply_mask` also returns a third element with `num_low_frequencies`.